tests/Http/HttpClientTest.php

```diff
@@ -223,4 +223,70 @@ public function testWithCookies()
         $this->assertSame('bar', $responseCookie['Value']);
         $this->assertSame('https://laravel.com', $responseCookie['Domain']);
     }
+
+    public function testGetWithArrayQueryParam()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get', ['foo' => 'bar']);
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?foo=bar';
+        });
+    }
+
+    public function testGetWithStringQueryParam()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get', 'foo=bar');
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?foo=bar';
+        });
+    }
+
+    public function testGetWithQuery()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get?foo=bar&page=1');
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?foo=bar&page=1';
+        });
+    }
+
+    public function testGetWithQueryWontEncode()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get?foo;bar;1;5;10&page=1');
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?foo;bar;1;5;10&page=1';
+        });
+    }
+
+    public function testGetWithArrayQueryParamOverwrites()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get?foo=bar&page=1', ['hello' => 'world']);
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?hello=world';
+        });
+    }
+
+    public function testGetWithArrayQueryParamEncodes()
+    {
+        $this->factory->fake();
+
+        $this->factory->get('http://foo.com/get', ['foo;bar; space test' => 'laravel']);
+
+        $this->factory->assertSent(function (Request $request) {
+            return $request->url() === 'http://foo.com/get?foo%3Bbar%3B%20space%20test=laravel';
```

Comment:
It would be nice to be able to get query params using a method. :thinking: should I do it? @taylorotwell

Reply:
I realized it can be done with `$request->data()['foo'] == 'bar'`. My bad.